R/MsBackendDataFrame-functions.R

```diff
@@ -75,9 +83,11 @@ NULL
 }
 
 .valid_intensity_mz_columns <- function(x) {
-    if (any(lengths(mz(x)) != lengths(intensity(x))))
-        "Length of mz and intensity values differ for some spectra"
-    else NULL
+    ## Don't want to have that tested on all on-disk objects.
+    if (length(x$intensity) && length(x$mz))
```

Comment:
In `BackendMzR` and `BackendHdf5Peaks` they don't exist. But even in `MsBackendDataFrame` they don't necessarily have to be present. You can create a backend without specifying the `mz` and `intensity` values. 

The idea is that you have these *core* spectra variables that ideally should be present and allow to define a spectrum, but you don't initialize the column if there is no value to set (no column instead of having a column with `NA`). Helps save memory and, even more importantly, we can later add additional core spectrum variables without making previously created object invalid, just because they don't contain the column (see https://github.com/rformassspectrometry/Spectra/issues/12#issuecomment-484423562)